apps/admin-ui/src/spa/reservations/[id]/index.tsx

```diff
@@ -524,169 +385,46 @@ function RequestedReservation({
           }
         />
       </ShowWhenTargetInvisible>
+
       <ReservationTitleSection
         ref={ref}
         reservation={reservation}
         tagline={reservationTagline}
       />
       <ButtonContainer $justifyContent="flex-start">
-        <ButtonsWithPermChecks
+        <ApprovalButtonsWithPermChecks
           reservation={reservation}
           onReservationUpdated={refetch}
           isFree={!isNonFree}
         />
       </ButtonContainer>
+
       <ReservationSummary reservation={reservation} isFree={!isNonFree} />
+
       <div>
-        <VisibleIfPermission
-          permission={UserPermissionChoice.CanViewReservations}
+        <ReservationWorkingMemoAccordion
           reservation={reservation}
-        >
-          <Accordion
-            id="reservation__working-memo"
-            heading={t("RequestedReservation.workingMemo")}
-            initiallyOpen={
-              reservation.workingMemo?.length != null &&
-              reservation.workingMemo?.length > 0
-            }
-          >
-            <ReservationWorkingMemo
-              reservationPk={reservation.pk ?? 0}
-              refetch={refetch}
-              initialValue={reservation.workingMemo ?? ""}
-            />
-          </Accordion>
-        </VisibleIfPermission>
+          onReservationUpdated={refetch}
+        />
+
         {reservation.accessType === AccessType.AccessCode && (
```

Comment:
If the reservation belongs to a recurringReservation, the check should be `recurringReservation.usedAcessTypes.includes(AccessType.AccessCode`) -- since the `usedAcessTypes` array contains all unique access types across its individual reservations and we wan't to show the section in the UI if any one them (potentially) uses a code.